src/Version/Version.cpp

```diff
@@ -89,8 +89,16 @@ std::string getOnnxMlirFullVersion(bool toIncludeLLVM) {
 #endif
   std::string repo = getOnnxMlirFullRepositoryVersion(toIncludeLLVM);
   if (!repo.empty()) {
-    os << " " << repo;
+    os << " " << repo << '\n';
   }
+#ifdef ONNX_MLIR_PRODUCT_VERSION
+  os << "onnx-mlir product version " ONNX_MLIR_PRODUCT_VERSION;
```

Comment:
Since this is to be used by vendors, we should make come before the rest of the version information. So something like:
```
onnx-mlir product version 4.0.0-zdlc, onnx-mlir version 0.3.1, onnx version 1.11.0 (https://github.com/hamptonm1/onnx-mlir.git 49f1f1e1db380b69abfdeb9ce455e8b99e0205fb)
LLVM version 16.0.0git (https://github.com/llvm/llvm-project.git 59548fe873d8d98e359fb21fbb2a0852fed17ff5)
DEBUG build with assertions.

Reply:
Sure! I can move that around. I was just following the output of the compiler which has onnx-mlir version first than the product version.

Reply:
@cjvolzka The only thing is having onnx-mlir version and onnx-version with the GitHub url is extremely long.... I think adding onnx-mlir product version on the same line will render awkwardly. I can test it out and see but I know content will probably be entered on the following line.

Reply:
Okay this is what it looks like:

```
root@d7f7476a247e:~/onnx-mlir/build# onnx-mlir --version
IBM Z Deep Learning Compiler 4.0.0-zdlc, onnx-mlir version 0.3.1, onnx version 1.11.0 (https://github.com/hamptonm1/onnx-mlir.git 40cdc98e1f4062fea9cf84e6a57c01dbef93f4c2)
 LLVM version 16.0.0git (https://github.com/llvm/llvm-project.git 59548fe873d8d98e359fb21fbb2a0852fed17ff5)
 DEBUG build with assertions.
 Default target: x86_64-unknown-linux-gnu
 Host CPU: skylake
```

As you can see the GitHub url is appended to the next line, so I am thinking what about this instead:

```
root@d7f7476a247e:~/onnx-mlir/build# onnx-mlir --version
IBM Z Deep Learning Compiler 4.0.0-zdlc
onnx-mlir version 0.3.1, onnx version 1.11.0 (https://github.com/hamptonm1/onnx-mlir.git40cdc98e1f4062fea9cf84e6a57c01dbef93f4c2)
LLVM version 16.0.0git (https://github.com/llvm/llvm-project.git 59548fe873d8d98e359fb21fbb2a0852fed17ff5)
DEBUG build with assertions.
Default target: x86_64-unknown-linux-gnu
Host CPU: skylake
```

Reply:
I'd be good with that. I'd actually be fine with putting onnx-mlir and onnx also on their own lines too. 

I'm unsure if that was discussed in the original PR that added VENDOR/VERSION support and rejected or just not brought up then.

Reply:
Okay sounds good to me!